geocoder/postal-codes/sql/geocoder.sql

```diff
@@ -27,4 +163,157 @@ CREATE OR REPLACE FUNCTION geocode_postalcode_polygons(code text[], inputcountri
   END LOOP;
   RETURN;
 END
-$$ LANGUAGE 'plpgsql' SECURITY DEFINER;
\ No newline at end of file
+$$ LANGUAGE 'plpgsql' SECURITY DEFINER;
+
+
+-- codes array integers, countries array
+CREATE OR REPLACE FUNCTION geocode_postalcode_polygons(code integer[], inputcountries text[])
+  RETURNS SETOF geocode_postalint_country_v1 AS $$
+  DECLARE 
+   ret geocode_postalint_country_v1%rowtype;
+BEGIN
+  FOR ret IN
+    SELECT
+      q, c, geom, CASE WHEN geom IS NULL THEN FALSE ELSE TRUE END AS success
+    FROM (
+      SELECT 
+        q, c, (
+          SELECT the_geom
+          FROM global_postal_code_polygons
+          WHERE postal_code_num = d.q
+            AND iso3 = (
+                SELECT iso3 FROM country_decoder WHERE
+                lower(d.c) = ANY (synonyms) LIMIT 1
+            )
+        ) geom
+      FROM (SELECT unnest(code) q, unnest(inputcountries) c) d
+    ) v
+  LOOP 
+    RETURN NEXT ret;
+  END LOOP;
+  RETURN;
+END
+$$ LANGUAGE 'plpgsql' SECURITY DEFINER;
+
+-- codes array text
+CREATE OR REPLACE FUNCTION geocode_postalcode_polygons(code text[])
+  RETURNS SETOF geocode_namedplace_v1 AS $$
+   DECLARE 
+    ret geocode_namedplace_v1%rowtype;
+  BEGIN
+  FOR ret IN
+    SELECT
+      q, geom, CASE WHEN geom IS NULL THEN FALSE ELSE TRUE END AS success
+    FROM (
+      SELECT 
+        q, (
+          SELECT the_geom 
+          FROM global_postal_code_polygons
+          WHERE postal_code = upper(d.q) 
+          LIMIT 1
+        ) geom
+      FROM (SELECT unnest(code) q) d
+    ) v
+  LOOP 
+    RETURN NEXT ret;
+  END LOOP;
+  RETURN;
+END
+$$ LANGUAGE 'plpgsql' SECURITY DEFINER;
+
+-- codes array text, countries text
+CREATE OR REPLACE FUNCTION geocode_postalcode_polygons(code text[], inputcountry text)
+  RETURNS SETOF geocode_namedplace_v1 AS $$
+  DECLARE 
+    ret geocode_namedplace_v1%rowtype;
+  BEGIN
+  FOR ret IN
+    SELECT
+      q, geom, CASE WHEN geom IS NULL THEN FALSE ELSE TRUE END AS success
+    FROM (
+      SELECT 
+        q, (
+          SELECT the_geom 
+          FROM global_postal_code_polygons
+          WHERE postal_code = upper(d.q)
+            AND iso3 = (
+                SELECT iso3 FROM country_decoder WHERE
+                lower(inputcountry) = ANY (synonyms) LIMIT 1
+            )
+        ) geom
+      FROM (SELECT unnest(code) q) d
+    ) v
+  LOOP 
+    RETURN NEXT ret;
+  END LOOP;
+  RETURN;
+END
+$$ LANGUAGE 'plpgsql' SECURITY DEFINER;
+
+
+-- Test functions for polygons --
+
+-- code array, countries array
+
+CREATE OR REPLACE FUNCTION test_geocode_postalcode_polygons(code text[], inputcountries text[])
```

Comment:
As far as I know, these *test* functions are the "new ones that were going to be put in production after testing", as all the geocoder functions were going to be rewritten but this process never finished.